src/main/java/core/basesyntax/model/Operation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.model;
+
+public enum Operation {
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private final String code;
+
+    Operation(String code) {
+        this.code = code;
+    }
+
+    public String getCode() {
+        return code;
+    }
+
+    public static Operation getByCode(String code) {
+        for (Operation operation : values()) {
+            if (operation.getCode().equals(code)) {
+                return operation;
+            }
+        }
+        throw new InvalidOperationCodeException("Invalid operation code: " + code);
+    }
+}

```

Comment:
The enum `Operation` is well implemented and follows the checklist guidelines. However, make sure to handle the case where `code` is `null` in the `getByCode` method to prevent a `NullPointerException`.